mcs/class/System.Core/System.IO.MemoryMappedFiles/MemoryMappedFile.cs

```diff
@@ -93,8 +93,17 @@ internal static void Map (IntPtr handle, long offset, ref long size, MemoryMappe
 			}
 		}
 
+		static void CheckString (string name, string value)
+		{
+			// Native code tends to truncate at nul which is incorrect. Guard it here.
+			if (value != null && value.IndexOf((char)0) != -1)
```

Comment:
- use `?.` syntax
- use `public ArgumentException (string message, string paramName) overload`
- use uppercase `NUL` name

Reply:
I find ?. really confusing. I tried to think of how it would be used here. I'll try again and agreed on the others. And maybe write tests.

Reply:
`if (value?.IndexOf((char)0) >= 0)` would be one way, but do we really want to accept `null` as a valid value?

Reply:
I still find this ?. highly unintuitive. What does it do when it recieves null??
It appears I get a nullable int "int?", not a regular "int", and then what operations does that have, how does null int? compare to things?

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/using-nullable-types

hen you perform comparisons with nullable types, if the value of one of the nullable types is null and the other is not, all comparisons evaluate to false except for != (not equal). It is important not to assume that because a particular comparison returns false, the opposite case returns true. In the following example, 10 is not greater than, less than, nor equal to null. Only num1 != num2 evaluates to true.
C#

int? num1 = 10;
int? num2 = null;
if (num1 >= num2)
{
    Console.WriteLine("num1 is greater than or equal to num2");
}
else
{
    // This clause is selected, but num1 is not less than num2.
    Console.WriteLine("num1 >= num2 returned false (but num1 < num2 also is false)");
}

if (num1 < num2)
{
    Console.WriteLine("num1 is less than num2");
}
else
{
    // The else clause is selected again, but num1 is not greater than
    // or equal to num2.
    Console.WriteLine("num1 < num2 returned false (but num1 >= num2 also is false)");
}

Reply:
I really do not find this clear.

commit 4d1f2b0d8242ed2f0079ec4cd216f29009ac2f52 (HEAD -> do-not-silently-truncate-at-nul, origin/do-not-silently-truncate-at-nul)
Author: Jay Krell <jaykrell@microsoft.com>
Date:   Wed Mar 21 15:55:47 2018 -0700

    Handle null correctly.

diff --git a/mcs/class/System.Core/System.IO.MemoryMappedFiles/MemoryMappedFile.cs b/mcs/class/System.Core/System.IO.MemoryMappedFiles/MemoryMappedFile.cs
index cf9cfd997d6..a35f74972f3 100644
--- a/mcs/class/System.Core/System.IO.MemoryMappedFiles/MemoryMappedFile.cs
+++ b/mcs/class/System.Core/System.IO.MemoryMappedFiles/MemoryMappedFile.cs
@@ -101,7 +101,7 @@ namespace System.IO.MemoryMappedFiles
                static void CheckString (string name, string value)
                {
                        // Native code tends to truncate at NUL which is incorrect. Guard it here.
-                       if (value?.IndexOf((char)0) != -1)
+                       if (value?.IndexOf((char)0) >= 0)
                                throw new ArgumentException ("String must not contain embedded NULs.", name);
                }